### ISSUE TITLE
CNTK write for speech example

### ISSUE THREAD

**Issue by vidyatshetty** [Microsoft/CNTK#2440]
How do i dump the values of labels in speech example?

**Comment by liwchang** [comment]
Maybe tutorial 204 and 106A might be close examples.

**Comment by vidyatshetty** [comment]
Thank you for writing @liwchang . I am basically using cntk to build a speaker recognition system. So far I have used HTK toolkit for data preparation and I created a configfile in cntk for training and testing the model. I get performance results too. Now I am actually looking to output the speaker ids from node outputs and I am struck here. Could you please let me know how do i proceed after training the model to get the output results(speaker id).
I am struggling with this from many days any information from you would be a great help for me.
Thanks.

**Comment by liwchang** [comment]
hmm, I am not sure I fully understand the key problem you are facing. 

After having the output vector through **eval**, typically people perform **softmax** (if it is not performed in the last layer yet) and then **argmax** to get the predicted label.
Or you can write your own fancy mapping from output vector to label. 

In Python,  softmax is in cntk.softmax, and argmax is in numpy.

**Comment by vidyatshetty** [comment]
I used the tutorial https://msdn.microsoft.com/en-us/magazine/mt791798 to understand how cntk works. In this particular case I get TestProbs_txt.pn file where i will be able to see the probability of the output labels. I am kind of trying to follow the same tutorial for my speaker recognition application aswell. I used the Brainscript cofigfile and gave feature and label files and got the performance results on CMD. Now I am want to generate a file which gives either probabilities of the output label and output label.
Is it possible to follow the https://msdn.microsoft.com/en-us/magazine/mt791798 tutorial for speech example aswell? or do i need to do it using python? Please let me know.

**Comment by liwchang** [comment]
Oops, I didn't realize you use BrainScript, my bad.
In the example your provided, I think it already called softmax in the end, so your output is a vector of probability.
Now, you just need an argmax, 
But I am even not sure whether BrainScript has a built-in argmax, most likely not.

If not, for work-around, I will suggest you do post-processing (by writing another program, say in python, to read the vector of probability, and perform argmax).

If you just started using CNTK (haven't invested a lot in BrainScript), I will suggest you use Python/C++/C# interface, instead of BrainScript.
In these three, outputs are stored in memory instead of files, and make post-processing. like argmax, can be easily perform in native python/C++/C#.

**Comment by vidyatshetty** [comment]
@liwchang thanks alot. I will follow your instructions and start to work using python. 
This really helped me.

**Comment by cha-zhang** [comment]
Most likely this is before Softmax. If you used cross_entropy_with_softmax, during testing you need to run softmax yourself to the output.

**Comment by vidyatshetty** [comment]
[testMlfOne.txt](https://github.com/Microsoft/CNTK/files/1419569/testMlfOne.txt)
@cha-zhang  Thanks for writing. The output I am getting is features. 
I used htk toolkit and completed data preparation steps(i have .scp files and label files)
I used cntk and gave these input files and when I output the results using htkmlfwriter I got feature dimension data.
I am confused why the network output is feature again. If i am not wrong I am using cntk to train the model using feature and label files and I am expecting label for the given Test feature file. I used network dimensions as 492:512:3:463
where 492 is input
512:3 is hidden hidden layers and units
463 is number of speaker

In the MLF file I have used label(here label is the speaker id) for each feature file. Is there anyway I can return the label if I give feature file for testing? I have used htk toolkit to build speaker recognition system and in htk when I give feature file for testing it outputs .mlf file. Is there a way I can expect CNTK to output labels similar to htk?
I am confused. 
Please let me know how do i use cntk to get labels as output.
I am attaching few samples of 
[testingOne.txt](https://github.com/Microsoft/CNTK/files/1419562/testingOne.txt)
 the .mlf file and feature file
